File: MIDIJava/src/io/github/seanboyy/Midi/SysexEvent.java
package io.github.seanboyy.Midi;

import java.util.ArrayList;

public class SysexEvent implements Event{
    private byte eventType;
    private VariableLengthValue timeDelta;
    private ArrayList<Byte> eventData;
	private VariableLengthValue eventLength;

	public SysexEvent(VariableLengthValue timeDelta, byte eventType, VariableLengthValue eventLength, ArrayList<Byte> eventData){
		this.timeDelta = new VariableLengthValue(timeDelta);
		this.eventType = eventType;
		this.eventLength = new VariableLengthValue(eventLength);
		this.eventData = new ArrayList<>();
		this.eventData.addAll(eventData);
	}

	@Override
	public byte[] toBitString(){
		byte[] bitString = new byte[getLength()];
		int[] cursor = {0};
		bitString = StreamingHelper.streamObjectToBitString(cursor, bitString, timeDelta, timeDelta.getLength());
		bitString = StreamingHelper.streamValueToBitString(cursor, bitString, eventType);
        bitString = StreamingHelper.streamObjectToBitString(cursor, bitString, eventLength, eventLength.getLength());
        for(Byte b : eventData){
            bitString = StreamingHelper.streamValueToBitString(cursor, bitString, b);
        }
        return bitString;
    }

	@Override
	public int getLength(){
		return timeDelta.getLength() + 1 + eventLength.getLength() + eventData.size();
	}
}

File: MIDIC#/MIDILibrary/MIDILibrary/SysexEvent.cs
using System.Collections.Generic;

namespace MIDILibrary
{
    public class SysexEvent : IEvent
    {
        public byte EventType { get; }
        public VariableLengthValue TimeDelta { get; }
        public List<byte> EventData { get; }
        public VariableLengthValue EventLength { get; }

        public SysexEvent(VariableLengthValue timeDelta, byte eventType, VariableLengthValue eventLength, List<byte> eventData)
        {
            TimeDelta = new VariableLengthValue(timeDelta);
            EventType = eventType;
            EventLength = new VariableLengthValue(eventLength);
            EventData = new List<byte>();
            EventData.AddRange(eventData);
        }

        public byte[] ToBitString()
        {
            byte[] bitString = new byte[GetLength()];
            uint cursor = 0;
            StreamingHelper.StreamObjectToBitString(ref cursor, ref bitString, TimeDelta, TimeDelta.Length);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, EventType);
            StreamingHelper.StreamObjectToBitString(ref cursor, ref bitString, EventLength, EventLength.Length);
            foreach (byte b in EventData)
            {
                StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, b);
            }
            return bitString;
        }

        public uint GetLength()
        {
            return TimeDelta.Length + 1 + EventLength.Length + (uint)EventData.Count;
        }
    }
}
